src/components/application_manager/src/resumption/resumption_data_db.cc

```diff
@@ -1006,6 +1016,59 @@ bool ResumptionDataDB::SelectSubscriptionsData(
   return true;
 }
 
+bool ResumptionDataDB::SelectUserLocationData(
+    const std::string& policy_app_id,
+    const std::string& device_id,
+    smart_objects::SmartObject& saved_app) const {
+  SDL_LOG_AUTO_TRACE();
+  using namespace app_mngr;
+  using namespace smart_objects;
+  uint32_t count_item = 0;
+  if (!SelectCountFromArray(
+          count_item, kSelectCountUserLocation, policy_app_id, device_id)) {
+    return false;
+  }
+
+  saved_app[strings::user_location] = SmartObject(SmartType_Map);
+
+  if (0 == count_item) {
+    SDL_LOG_INFO("Application does not contain user_location data");
+    return true;
+  }
+  utils::dbms::SQLQuery select_user_location(db());
+  if (!PrepareSelectQuery(select_user_location,
+                          policy_app_id,
+                          device_id,
+                          kSelectUserLocation)) {
+    SDL_LOG_WARN("Problem with verification user_location");
+    return false;
+  }
+  if (!select_user_location.Exec()) {
+    SDL_LOG_WARN("Problem with execution select_user_location query");
+    return false;
+  }
+  /* Position of data in "select_user_location" :
```

Comment:
This comment may be redundant, because this order is fixed in kSelectUserLocation constant, isn't it?

Reply:
@ychernysheva order is fixed, but this comment can be kept just for explaining of magic numbers in this query so it is mostly for readability purposes